### DISCUSSION TITLE
Styling cells based on their content

### DISCUSSION THREAD

**Discussion by dhicks** [rstudio/gt#1779] [Show and tell]
Context: I'm reporting the results of some exploratory factor analyses, and in the loadings table I need to **bold** cells whose (absolute) values are greater than 0.3.  

It's surprising (to me!) how difficult it is to get gt to style cells individually based on their content across multiple columns. There was a [feature request](https://github.com/rstudio/gt/issues/703) to facilitate in December 2020; it was closed almost two years later when [`sub_values()`](https://github.com/rstudio/gt/pull/1066) was added.  But `sub_values()` is for manipulating the content of cells, not applying styling, so it's not clear to me why this justified closing the issue. 

The standard solution still appears to be [writing a clunky for loop](https://stackoverflow.com/questions/63944953/how-to-conditionally-format-a-cell-in-a-gt-table-based-on-the-value-of-the-cel/63945239#63945239).  I spent a little time this morning converting that into a more flexible function that can be placed in a pipe. MWE below. 

```# Load necessary libraries
library(gt)
library(dplyr)
library(rlang)

# Create a sample data frame and a gt table
data <- data.frame(
    category = c("A", "B", "C"),
    value1 = c(123, 234, 345),
    value2 = c(456, 567, 678),
    value3 = c(789, 890, 901)
)

gt_table <- gt(data)
gt_table

## The star of the show
## this_gt: the gt object
## col_select: tidy-select pattern for the columns of interest
## cell_select: formula-style function with the condition for highlighted/styled cells
## cell_style: styling to be applied to the cells picked out by cell_select
highlight_cells = function(this_gt, 
                           col_select = everything(), 
                           cell_select = ~ . > 500, 
                           cell_style = cell_text(weight = 'bold')) {
    ## Identify columns
    cols = this_gt$`_data` |> 
        select({{ col_select }}) |> 
        colnames()
    
    ## Convert the cell selector formula into a function
    cell_selector <- rlang::as_function(cell_select)
    
    ## Iterate columnwise, applying the styling using the cell selector
    for (col in cols) {
        this_gt <- this_gt |> 
            tab_style(
                style = cell_style,
                locations = cells_body(
                    columns = all_of(col),
                    rows = cell_selector(.data[[col]]))
            )
    }
    return(this_gt)
}

highlight_cells(gt_table)
highlight_cells(gt_table, where(is.numeric), cell_select = ~ . > 600)
```

**Comment by rich-iannone** [comment]
Within the gt API there is the `tab_style_body()` function (https://gt.rstudio.com/reference/tab_style_body.html). Take a look at the documentation and let me know if this works well for your use case!
